src/sos/section_analyzer.py

```diff
@@ -398,7 +398,7 @@ def get_step_input(section, default_input):
                 step_input = sos_targets()
             else:
                 step_input = default_input
-        elif not any(isinstance(x, (dynamic, remote)) for x in args):
+        elif not any(isinstance(x, (dynamic, remote)) for x in kwargs):
```

Comment:
You changed the logic of the code, which is wrong. I do not know why there is a warning here but `args` was indeed defined in line 394.